### PR TITLE
*: enforce errcheck in Makefile

### PR DESCRIPTION
**Pull request by coocood**
checks all errors and enable errcheck in Makefile

Fixes https://github.com/pingcap/tidb/issues/4668

### CODE REVIEW

File: expression/util.go [hunk 1]
```diff
@@ -236,7 +239,7 @@ func PushDownNot(expr Expression, not bool, ctx context.Context) Expression {
 				for i, a := range args {
 					args[i] = PushDownNot(a, true, f.GetCtx())
 				}
-				nf, _ := NewFunction(f.GetCtx(), ast.LogicAnd, f.GetType(), args...)
+				nf := NewFunctionInternal(f.GetCtx(), ast.LogicAnd, f.GetType(), args...)
```
Comment (shenli) [inline expression/util.go]: return NewFunctionInternal(f.GetCtx(), ast.LogicAnd, f.GetType(), args...)

File: expression/util.go [hunk 2]
```diff
@@ -61,7 +62,7 @@ func ColumnSubstitute(expr Expression, schema *Schema, newExprs []Expression) Ex
 		for _, arg := range v.GetArgs() {
 			newArgs = append(newArgs, ColumnSubstitute(arg, schema, newExprs))
 		}
-		fun, _ := NewFunction(v.GetCtx(), v.FuncName.L, v.RetType, newArgs...)
+		fun := NewFunctionInternal(v.GetCtx(), v.FuncName.L, v.RetType, newArgs...)
```
Comment (shenli) [inline expression/util.go]: return NewFunctionInternal(v.GetCtx(), v.FuncName.L, v.RetType, newArgs...)

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
@shenli PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**